engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -18,7 +18,7 @@
 public class ModuleScope implements TruffleObject {
   private final AtomConstructor associatedType;
   private final Module module;
-  private Map<String, Object> polyglotSymbols = new HashMap<>();
+  private final Map<String, Object> polyglotSymbols = new HashMap<>();
```

Comment:
The fact that you can make this final indicates a probable oversight. I suspect it should be reassigned in `reset()`, but please ensure that the tests pass after doing so.